proxy/src/proxy.rs

```diff
@@ -966,8 +978,46 @@ impl<S: AsyncRead + AsyncWrite + Unpin> Client<'_, S> {
             params,
             session_id,
             allow_self_signed_compute,
+            endpoint_rate_limiter,
         } = self;
 
+        // Simple rate limiter.
+        //
+        // Check that number of connections to that endpoint is below 300 rps
+        // (purposefully ignoring user name and database names).
```

Comment:
> purposefully ignoring user name and database names

I'd add the reasoning why: (iirc) because if we don't, then someone could circumvent rate limiting by connecting with random users.